src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -385,14 +385,28 @@ private static function analyzeNonDivOperands(
 
         if ($left_type_part instanceof Type\Atomic\TString
             && $right_type_part instanceof TInt
-            && $parent instanceof PhpParser\Node\Expr\PostInc
+            && (
+                $parent instanceof PhpParser\Node\Expr\PostInc ||
+                $parent instanceof PhpParser\Node\Expr\PreInc
+            )
         ) {
-            $has_string_increment = true;
+            if ($left_type_part instanceof Type\Atomic\TNumericString || (
+                    $left_type_part instanceof Type\Atomic\TLiteralString &&
+                    //Matches '1', '1.', '-1', '-1.'. Does not match '.1'
+                    preg_match('/^\-?\d+\.?\d*$/', $left_type_part->value)
```

Comment:
> Does not match '.1'

It does work with increment though: https://3v4l.org/updn7

It isn't terribly important to support that, of course.

Reply:
Doesn't increment work on any string that is `is_numeric()`? If so, and given that Psalm doesn't run on PHP 5 where `is_numeric()` also accepted hex-like strings, can't we just replace that with `is_numeric($left_type_part->value)`?

Reply:
Yeah, that's why I stated it explicitly, I didn't saw how I could do a regex to support all that and I didn't want to complexify too much. There is little chance for finding this in the wild so it will stay like this for now :)

Reply:
I guess we can. I stayed out of it because I don't really master all the subtleties of is_numeric but I don't see harm in it as long as we don't try to guess the exact Literal it generates